### PR TITLE
[TOAZ-37] Add Azure Relay to CRL

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/TOAZ-37

Depends on https://github.com/DataBiosphere/terra-resource-janitor/pull/127

Janitor version is a locally published version just to get the a few swagger generated azure models...Will have a follow up PR to update janitor's version once the janitor PR is in

cc @sergiygetlin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,6 +159,9 @@ Artifact Repository can be found [here](https://broadinstitute.jfrog.io/broadins
 #### azure-resourcemanager-compute
 Wraps [Azure Compute API](https://docs.microsoft.com/en-us/rest/api/compute/).
 Artifact Repository can be found [here](https://broadinstitute.jfrog.io/broadinstitute/webapp/#/artifacts/browse/tree/General/libs-snapshot-local/bio/terra/cloud-resource-lib/azure-resourcemanager-compute).
+#### azure-resourcemanager-compute
```
Comment (yonghaoy) [inline README.md:162]: azure-resourcemanager-**relay**

File: azure-resourcemanager-relay/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Please update platform to consume version updates, see here for more:
+# https://github.com/DataBiosphere/terra-cloud-resource-lib#publishing-an-update
+version = 0.1.2
```
Comment (yonghaoy) [inline azure-resourcemanager-relay/gradle.properties:3]: that is the version for this package. Why does it start from 0.1.2?

File: azure-resourcemanager-relay/src/main/java/bio/terra/cloudres/azure/resourcemanager/relay/Defaults.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     * <pre>
+     *    crlConfigure(clientConfig, ComputeManager.configure())
+     *        .authenticate(tokenCredential, azureProfile);
+     * </pre>
+     *
+     * @param clientConfig client configuration object to manage CRL behavior.
+     * @param configurable Azure client to configure.
+     * @return a configured Azure client.
+     */
+    public static Configurable crlConfigure(
```
Comment (yonghaoy) [inline azure-resourcemanager-relay/src/main/java/bio/terra/cloudres/azure/resourcemanager/relay/Defaults.java:25]: I throught common/Defaults is a common code that can be used for all `azure/resourcedmanager`. Why do we need the new one? Can we just use 
https://cs.github.com/DataBiosphere/terra-cloud-resource-lib/blob/a13ed003bc6a7b31681a4022a4fa15606b416159/azure-resourcemanager-common/src/main/java/bio/terra/cloudres/azure/resourcemanager/common/Defaults.java?q=AzureResourceCleanupRecorder#L64?

File: build.gradle [hunk 4]
```diff
@@ -39,6 +39,7 @@ artifactory {
                     'google-storage',
                     'platform',
                     'azure-resourcemanager-common',
+                    'azure-resourcemanager-relay',
```
Comment (yonghaoy) [inline build.gradle:42]: nit: alphabetically sort is nicer, move `azure-resourcemanager-relay` below `azure-resourcemanager-compute`?

File: azure-resourcemanager-relay/gradle.properties (hunk 2 above)
Comment (Qi77Qi) [reply azure-resourcemanager-relay/gradle.properties:3]: I was trying to publish it locally, so I updated a few times..I'll change it back

File: azure-resourcemanager-relay/src/main/java/bio/terra/cloudres/azure/resourcemanager/relay/Defaults.java (hunk 3 above)
Comment (rtitle) [reply azure-resourcemanager-relay/src/main/java/bio/terra/cloudres/azure/resourcemanager/relay/Defaults.java:25]: That's unfortunate, I thought all client libraries would use `AzureConfigurable` 😞

**Review by yonghaoy — APPROVED** [review]

File: azure-resourcemanager-relay/src/main/java/bio/terra/cloudres/azure/resourcemanager/relay/Defaults.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import com.azure.resourcemanager.relay.RelayManager.Configurable;
+
+public class Defaults {
+    /**
+     * Configures a client for CRL usage.
+     *
+     * <p>Example usage:
+     *
+     * <pre>
+     *    crlConfigure(clientConfig, ComputeManager.configure())
```
Comment (rtitle) [inline azure-resourcemanager-relay/src/main/java/bio/terra/cloudres/azure/resourcemanager/relay/Defaults.java:17]: Maybe update the example to use `RelayManager`

File: google-api-services-common/gradle.properties [hunk 6]
```diff
@@ -1,3 +1,3 @@
 # Please update platform to consume version updates, see here for more:
 # https://github.com/DataBiosphere/terra-cloud-resource-lib#publishing-an-update
-version = 0.10.0
+version = 0.11.0
```
Comment (rtitle) [inline google-api-services-common/gradle.properties:3]: Did you intend to version bump this module?

File: buildSrc/src/main/groovy/terra-cloud-resource-lib.library-conventions.gradle [hunk 7]
```diff
@@ -37,7 +37,7 @@ configurations {
 
 dependencies {
     ext {
-        janitorclient = '0.5.0-SNAPSHOT'
+        janitorclient = '0.6.1-SNAPSHOT'
```
Comment (rtitle) [inline buildSrc/src/main/groovy/terra-cloud-resource-lib.library-conventions.gradle:40]: Technically I guess this change alters the dependencies for every CRL module… but I don't think it's really version bumping each CRL module for this. Just note that the CI artifactory push might fail after merge, but I think the new `azure-relay` should get pushed.

**Review by rtitle — APPROVED** [review]
Looks good

File: buildSrc/src/main/groovy/terra-cloud-resource-lib.library-conventions.gradle (hunk 7 above)
Comment (Qi77Qi) [reply buildSrc/src/main/groovy/terra-cloud-resource-lib.library-conventions.gradle:40]: ah good to know...yea, as long as relay is pushed, I'll update janitor next to get published relay. And then I can update janitor version here for real....

**Pull request merged**